### ISSUE TITLE
[Accessibility]: Focus/hover states on the menu dropdown

### ISSUE THREAD

**Issue by marisak** [sourcegraph/sourcegraph#35160]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34197

### Problem description

The focus state and hover state for the menu dropdown component appear to be the same. This is adequate for mouse interaction, but confusing for keyboard navigation. This can be replicated on any menu dropdown; the one I was particularly struggling with was in Notebooks, in the menu for deleting an individual notebook. 

### Expected behavior

I expected the active state to present as an outline, but it's just shown as a grey background on the item. This is especially confusing for the delete menu item, which currently has a red background. I kept exiting the dropdown because I couldn't tell that the grey background indicated the focused state. Potentially related to the [design debt ticket about delete buttons.](https://github.com/sourcegraph/sourcegraph/issues/34058) 

### Additional details

Regular state (red background): 
<img width="1440" alt="Screen Shot 2022-05-09 at 4 13 56 PM" src="https://user-images.githubusercontent.com/16570574/167490386-71fadebd-5373-4c3f-830a-f43c2ccf1826.png">
Focus state (grey background): 
<img width="1440" alt="Screen Shot 2022-05-09 at 4 16 31 PM" src="https://user-images.githubusercontent.com/16570574/167490387-73f80abc-06d1-4126-99f5-a0f81058adf3.png">

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [ ] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

I'm unsure who should work on this issue or if it needs more clarification. I will let the Frontend Platform team triage this.

**Comment by umpox** [comment]
Might be fixed by https://github.com/sourcegraph/sourcegraph/issues/36350
